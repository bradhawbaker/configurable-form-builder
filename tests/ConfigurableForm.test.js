import React from 'react';
import renderer from 'react-test-renderer';

import ConfiguraleForm from '../src/ConfigurableForm';

const SINGLE_TEXTFIELD_FORM_CONFIG = {
    title: "Sample Config Form Title",
    fields: [
      {
        id: "id",
        type: "text",
        label: "Name"
      },
      {
        id: "status",
        type: "enum",
        label: "Status",
        values: ["active", "error", "pending"]
      }
    ],
    primaryButtonText: "Search"
  };

test('Basic snapshot', () => {
    const component = renderer.create(
        <ConfiguraleForm 
            {...SINGLE_TEXTFIELD_FORM_CONFIG} 
            primaryButtonCallback={ () => {
                console.log('simple callback');
            }} 
        />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});