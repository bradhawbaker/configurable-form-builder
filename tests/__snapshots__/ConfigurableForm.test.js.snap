// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Basic snapshot 1`] = `
<div
  className="configurable-form-builder"
>
  <h1>
    Sample Config Form Title
  </h1>
  <form>
    <div>
      <label>
        Name
      </label>
      <input
        id="id"
        name="id"
        onChange={[Function]}
        type="text"
        value=""
      />
    </div>
    <div>
      <label>
        Status
      </label>
      <select
        defaultValue="noSelection"
        id="status"
        onChange={[Function]}
      >
        <option
          value=""
        >
           
        </option>
        <option
          value="active"
        >
          active
        </option>
        <option
          value="error"
        >
          error
        </option>
        <option
          value="pending"
        >
          pending
        </option>
      </select>
    </div>
    <button
      className="primary"
      onClick={[Function]}
    >
      Search
    </button>
  </form>
</div>
`;
